src/STON-Core/Interval.extension.st

```diff
@@ -12,6 +12,12 @@ Interval >> fromSton: stonReader [
 	stonReader parseNamedInstVarsFor: self
 ]
 
+{ #category : #'*STON-Core' }
+Interval >> sort: aBlock [
+	"Directly returns the Interval, since sorting an Interval does not make sense."
```

Comment:
I think we should not say it does not make sense to `#sort:` an `Interval` in the comment. Else, the method should simply not be implemented.

It may make sense that one can call `#sort:` on an interval but I'm not sure how it should be implemented.

For example, if the following evaluate correctly, it might be ok:
```
((1 to: 10) sorted: [ :a :b | a > b ]) >>> (10 to: 1 by: -1)
```

But in some case, depending on the sort block, an instance of `Interval` can not be returned.

Reply:
Interval is a subclass of SequenceableCollection which has a default implementation of #sort:
This default implementation results in an exception when used on an Interval because you cannot #at:put: in an Interval.
"does not make sense" is maybe a bit too much. I meant by that that an Interval is not a concept that has an intuitive notion of what it means to sort it.
So I implemented a trivial sort function to override the default one from SequenceableCollection to at least not raise a strange exception when someone sorts an Interval by accident.
I guess that the sort method of Interval could raise an explicit "Cannot sort an Interval" exception instead of the "you can not store into an interval" exception raised by Interval>>at:put:

Reply:
I made a new issue, branch and pull request for this: https://github.com/pharo-project/pharo/pull/1922
It's just adding the Interval>>sort: method and having it raise an "Can not sort an Interval" error.

Reply:
Ah, ok nervermind may comment on the other thread, I didn't know Sequenceable collection implement `#sort:`.